bazel/wasm/wasm.bzl

```diff
@@ -0,0 +1,54 @@
+def _wasm_transition_impl(settings, attr):
+    return {
+        "//command_line_option:cpu": "wasm",
+        "//command_line_option:crosstool_top": "@proxy_wasm_cpp_sdk//toolchain:emscripten",
+        # Overriding copt/cxxopt/linkopt to prevent sanitizers/coverage options leak into WASM build
+        "//command_line_option:copt": [],
+        "//command_line_option:cxxopt": [],
+        "//command_line_option:linkopt": [],
+    }
+
+wasm_transition = transition(
+    implementation = _wasm_transition_impl,
+    inputs = [],
+    outputs = [
+        "//command_line_option:cpu",
+        "//command_line_option:copt",
+        "//command_line_option:cxxopt",
+        "//command_line_option:linkopt",
+        "//command_line_option:crosstool_top",
+    ],
+)
+
+def _wasm_binary_impl(ctx):
+    out = ctx.actions.declare_file(ctx.label.name)
+    ctx.actions.run_shell(
+        command = 'cp "{}" "{}"'.format(ctx.files.binary[0].path, out.path),
+        outputs = [out],
+        inputs = ctx.files.binary,
+    )
+    print(ctx.files.binary[0].path)
+    print(out.path)
+
+    return [DefaultInfo(files = depset([out]))]
+
+wasm_binary = rule(
+    implementation = _wasm_binary_impl,
+    attrs = {
+        "binary": attr.label(mandatory = True, cfg = wasm_transition),
+        "_whitelist_function_transition": attr.label(default = "@bazel_tools//tools/whitelists/function_transition_whitelist"),
+    },
+)
+
+def wasm_cc_binary(name, **kwargs):
+    wasm_name = "_wasm_" + name
+    native.cc_binary(
+        name = wasm_name,
+        # Adding manual tag so it won't be built in non-wasm config automatically.
```

Comment:
It would be great if we could get the .wasm for non-wasm targets which depend on them.

Reply:
Yes, that's the `wasm_binary` below does, added more comment. This is the private cc_binary for the one below to depend on, so it has `_wasm_` prefix.